server/src/Application/Repositories/ILectureRepository.cs

```diff
@@ -4,4 +4,5 @@ namespace StudyZen.Application.Repositories;
 
 public interface ILectureRepository : IRepository<Lecture>
 {
-}
\ No newline at end of file
+    Task<List<Lecture>> GetLecturesByCourseId(int courseId);
```

Comment:
I think this is unnecessary since we have a foreign key (collection of lectures in course)

Reply:
`public async Task<IReadOnlyCollection<LectureDto>> GetLecturesByCourseId(int courseId)
    {
        var courseLectures = await _lectures.GetLecturesByCourseId(courseId);
        return courseLectures.Select(lecture => new LectureDto(lecture)).ToList();

    }`
so there you would suggest retrieving course with eager loading to get the lectures?